pkg/git/libgit2/managed/http.go

```diff
@@ -133,6 +153,16 @@ func (t *httpSmartSubtransport) Action(transportOptionsURL string, action git2go
 	}
 	t.httpTransport.DisableCompression = false
 
+	t.once.Do(func() {
+		if opts.Context != nil {
+			t.ctx = opts.Context
+			t.logger = logr.FromContextOrDiscard(t.ctx).
```

Comment:
I could be wrong but my understanding now is that the subtransport factory creates a subtransport with context as `context.Background()`. This context doesn't contain any logger, which would make the above line to return a discard logger.
Previously, when the `InitManagedTransport()` was called in main.go with a context, that also contained the controller-runtime logger and since we configured the logger at the beginning, the logger passed to `InitManagedTransport()` brought the log flag options set in main. I think all that's lost in this approach, as a new context is created per transport which doesn't inherit the main application's log configurations.

If my above thinking is true, `InitManagedTransport()` should accept a logger and save it in a variable. And in the subtransport factory, we can use `logr.NewContext(context.Background(), logger)` to inherit the global application log configuration in each of the transport loggers. And then in per Action(), can add its own url, cid, addr, etc values.

Reply:
One of the reasons I removed the logr sent on `InitManagedTransport()` was that it was blocking the initialisation via `init()`, which may be the best way of doing this once we decomission unmanaged transport.

From my local tests, it does feel like the returned logr instance honours the log configurations. I have updated the log example on the issue description with the latest changes.

Reply:
Looks like in `Action()`, it extracts the logger from the transport option context, which inherits the logger from checkout, which gets it from the reconciler, so that has the logger in the context, and it works. So, in the usual cases, it'll be fine.
Only when the transport option doesn't have a context with logger, then we'll see the issue, but that's not likely to happen. So no problem.